### ISSUE TITLE
yt broken: Unable to extract JS player URL

### ISSUE THREAD

**Issue by samiq** [rg3/youtube-dl#4792]
Hi guys,

I have the youtube-dl running as a web service on an app engine to which an an app I've built requests the urls of  yt videos people want to play so they can be streamed

from time to time, when google changes the signatures, this will stop working, I update with the new code and all keeps working

today the service stopped working, I don’t know if this has been caught as google changing the signatures or something else but the service is failing on me with 

*An error happened while fetching video '7Ya2U8XN_Zw' with: ERROR: Unable to extract JS player URL; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.*

now, I try to get the same video from my mac's terminal and it works just fine.

it's the same code running on both ends and it's the latest here in github, don't know if some can't help me debug this or has encountered something like this before

**Comment by samiq** [comment]
Hi guys, seems like if I change the video, I get a different error

[youtube] MYSVMgRr6pw: Downloading webpage
[youtube] MYSVMgRr6pw: Extracting video information
[youtube] MYSVMgRr6pw: Downloading DASH manifest
An error happened while fetching video 'MYSVMgRr6pw' with: invalid literal for int() with base 10: 'nondash-171'

**Comment by dstftw** [comment]
Post the **full** output you get when running [with `--verbose` option](https://github.com/rg3/youtube-dl/blob/master/README.md#bugs).

**Comment by dstftw** [comment]
`*.dump` files obtained with `--write-pages` would be helpful as well.

**Comment by samiq** [comment]
thing is that in my mac (UK) is working, but in the server is not (US) same code... now I just found out about this http://venturebeat.com/2015/01/27/youtube-ditches-flash-for-html5-video-by-default/ don't know if it's related and I am still getting the old html here in UK vs US. I will try to see how I can dump the pages from the server, since in app engine I can't do that

**Comment by samiq** [comment]
Ok guys was able to get the dump for here in UK and for the one the server is getting:

https://www.dropbox.com/sh/q0ei3la3ta8pzoa/AAAbI3u1CSJ0cDG_QfkpPs31a?dl=0

both files are different, the one I'm getting here in the uk has the assets, js line  that is searched for in line 1020 of the code as in 
```
jsplayer_url_json = self._search_regex(
                        r'"assets":.+?"js":\s*("[^"]+")',
                        embed_webpage if age_gate else video_webpage, 'JS player URL')
```
where the one the server is getting doesn't, so I'm guessing google did change the way html exists

appreciate the help and hope this info helps you guys fixing it

**Comment by jaimeMF** [comment]
Probably it's the same as #4717, in the page from US I see this message:
```html
<h1 id="unavailable-message" class="message">
    The uploader has not made this video available in your country.
</h1>
```

We should detect it and report an error (assuming that this is the message you actually get when accessing the page in a browser from the US).

**Comment by samiq** [comment]
I don't know. Everything was working fine and out of the blue this afternoon all videos went blank. So I don't think is the same issue, as it would have gone blank from before. Something changed in the pages google serve that broke the scripts. I can see it in the logs of my server

**Comment by phihag** [comment]
Notably, the error messages are a red herring. Can you provide a proxy at a problematic location? ([in private](mailto:phihag@phihag.de) is fine). Without a proxy, I don't think we'll be able to adapt the extractor.

**Comment by phihag** [comment]
I believe this issue to be fixed in youtube-dl 2015.01.30.2, where we look in the embed page as well. I'm assuming the new version fixes the problem for you, but if it doesn't (or even if it does), please report back. We will reopen this issue then if necessary. Thanks for the report!
